spark_log_parser/parsing_models/application_model.py

```diff
@@ -194,20 +195,26 @@ def __init__(self, eventlogpath, bucket=None, stdoutpath=None, debug=False):  #
 
                 elif event_type == "SparkListenerEnvironmentUpdate":
 
-                    curKeys = json_data["Spark Properties"].keys()
+                    spark_properties = json_data["Spark Properties"]
+                    event_keys = spark_properties.keys()
 
                     # This if is specifically for databricks logs
-                    if "spark.databricks.clusterUsageTags.sparkVersion" in curKeys:
-
+                    if spark_version := spark_properties["spark.databricks.clusterUsageTags.sparkVersion"]:
                         self.cloud_platform = "databricks"
-                        self.cloud_provider = json_data["Spark Properties"][
+                        self.spark_version = spark_version
+                        self.cluster_id = spark_properties["spark.databricks.clusterUsageTags.clusterId"]
+                        self.cloud_provider = spark_properties[
                             "spark.databricks.clusterUsageTags.cloudProvider"
                         ].lower()
-                        self.spark_version = json_data["Spark Properties"][
-                            "spark.databricks.clusterUsageTags.sparkVersion"
-                        ]
+                    elif cluster_id := json_data["System Properties"]["EMR_CLUSTER_ID"]:
+                        self.cloud_platform = "emr"
+                        self.cloud_provider = "aws"
+                        self.cluster_id = cluster_id
+                    else:
+                        raise UrgentEventValidationException(
+                            "Unable to determine the platform this job was run on - these logs may be from an unsupported platform.")
 
-                    self.spark_metadata = {**self.spark_metadata, **json_data["Spark Properties"]}
+                    self.spark_metadata = {**self.spark_metadata, **spark_properties}
```

Comment:
I felt it appropriate to fill in the `cloud_platform`/`provider` fields here, since we are now using this condition to pull EMR-specific metadata. If we don't have either of these pieces of data, we will presume that this is an un-recognized platform and raise an exception.

Reply:
It's possible that a `SparkListenerEnvironmentUpdate` event won't exist in an EMR log, so probably best to keep that determination at the end as it was.

Reply:
It's reliably there for Databricks, as far as I've seen

Reply:
Also, is the EMR version identified here? If so we should grab that too.